TypeCobol/Compiler/CodeElements/Expressions/Expression.cs

```diff
@@ -201,20 +243,81 @@ public override bool AcceptASTVisitor(IASTVisitor astVisitor)
 		}
 	}
 
-	/// <summary>
-	/// Relational operators and their meanings
-	/// </summary>
-	public enum RelationalOperator
-	{
-		EqualTo,
-		GreaterThan,
-		GreaterThanOrEqualTo,
-		LessThan,
-		LessThanOrEqualTo,
-		NotEqualTo
-	}
+    public class RelationalOperator : IVisitable
+    {
+        /// <summary>
+        /// If the operator is negated, Token of the NOT
+        /// Null if no NOT is present
+        /// </summary>
+        public Token NotToken { get; }
+        public SyntaxProperty<RelationalOperatorSymbol> Operator { get; }
+        
+        public RelationalOperator(SyntaxProperty<RelationalOperatorSymbol> @operator, Token notToken)
+        {
+            Operator = @operator;
+            NotToken = notToken;
+        }
 
-	/// <summary>
+        /// <summary>
+        /// Return a RelationalOperatorSymbol which has the semantic value of this object
+        /// </summary>
+        public RelationalOperatorSymbol SemanticOperator
+        {
+            get
+            {
+                if (NotToken == null)
+                {
+                    return Operator.Value;
+                }
+                switch (Operator.Value)
+                {
+                    case RelationalOperatorSymbol.EqualTo:
+                        return RelationalOperatorSymbol.NotEqualTo;
+                    case RelationalOperatorSymbol.NotEqualTo:
```

Comment:
`Operator.Value` should never be equal to `NotEqualTo` so this label should be removed.

Reply:
Ok, to clarify, a comment has been added to the `default` case.